_posts/2020-07-01-ansible-jak-uporzadkowac-chaos.md

```diff
@@ -0,0 +1,308 @@
+
+# Era Ansible ![Logo Ansible](/assets/img/posts/2020-07-01-ansible-jak-uporzadkowac-chaos/ansible.png)
+Już od wczesnych lat studiów programistom wpaja się, aby dążyć do utrzymania eleganckiej i czytelnej struktury swoich aplikacji (`KISS – Keep it simple, stupid`). Doskonale w ten trend wpasowuje się **Ansible**. To kupione przez firmę **Red Hat** opensource’owe oprogramowanie m.in. do automatyzacji procesu wdrażania aplikacji i zarządzania konfiguracją. W prosty sposób, za pomocą języka **YAML** pozwala opisać wzajemne relacje między systemami. 
+
+Czytając dokumentację dostajemy obietnicę ujednolicenia konfiguracji, organizacji złożonych procesów i jednocześnie łatwą do zarządzania architekturę. Ponadto czytamy, że Ansible pozwala osiągnąć wzrost wydajności i nie nakłada dodatkowych wymagań na otoczenie, w którym działa. 
+
+Do pracy z Ansible wymagane jest tylko, aby na maszynie sterującej zainstalowane były: 
+- Python 2.7 lub Python 3.5+
+- Ansible (np. przy użyciu Python Package Manager’a) 
+
+Ansible wymaga hasła lub klucza SSH w celu rozpoczęcia zarządzania systemami i może rozpocząć zarządzanie nimi bez instalowania oprogramowania agenta, unikając problemu „zarządzania zarządzaniem” powszechnego w wielu systemach do automatyzacji.
+
+Ansible łączy się z systemami poprzez mechanizmy transportowe – SSH (Unix) lub PowerShell (Windows). **Moduły**, które są małymi programami, zawierające uzupełnione argumenty, przenoszone są do tymczasowego katalogu przez wspomniane mechanizmy na zarządzane maszyny. Tam są wykonywane, a następnie usuwane w ramach jednej akcji. Moduły zwracają obiekty JSON na standardowe wyjście, a te z kolei przetwarzane są przez program Ansible na maszynie sterującej. Mogą one zarządzać idempotentnymi zasobami. Oznacza to, że moduł działa w sposób deklaratywny, czyli może zdecydować np. czy dany pakiet powinien zostać zainstalowany w określonej wersji lub nie wykonać żadnej akcji, gdy system jest już w pożądanym stanie. Można je też uruchamiać pojedynczo (imperatywnie).
+
+Do zarządzania administrowanymi systemami służą pliki **inventory**. Pozwalają one na grupowanie serwerów i definiowanie zmiennych, które później wykorzystywane będą w tzw. **playbookach**. Plik inventory może być użyty globalnie, gdzie zainstalowano Ansible lub wykorzystać lokalne pliki inventory dedykowane dla konkretnego projektu. Przykładowy fragment pliku inventory wygląda następująco:
+```
+### Ogólne parametry instalacji
+
+validation_protocol: http
+validation_host: eximee-validation
+validation_port: 8080
+services_protocol: http
+services_host: eximee-services
+services_port: 8080
+servicemix_host: eximee-servicemix
+# Poniższe "placeholdery" opakowane w nawiasy klamrowe nie powinny zawwierać spacji. 
+# Występuje ona tylko dlatego, że zmienne znajdujące się wewnątrz nawiasów nie są prawidłowo renderowane na stronie z wpisem.
```

Comment:
A nie da się wyescape'ować?

Reply:
{% raw %}
Hello, my name is {{name}}.
{% endraw %}

Reply:
super, dzięki :) poprawione